### PR TITLE
use-cilium-eni-mode

### PR DESCRIPTION
**Pull request by calvix**
configure EKS to use cilium in ENI mode
towards https://github.com/giantswarm/giantswarm/issues/28553

### CODE REVIEW

File: helm/cluster-eks/templates/cilium-helmrelease.yaml [hunk 1]
```diff
@@ -16,10 +16,10 @@ spec:
       chart: cilium
       # used by renovate
       # repo: giantswarm/cilium-app
-      version: 0.10.0
+      version: 0.14.0-2c233645daf0c4fad59f5553cf41f96ddfcc54bc
```
Comment (calvix) [inline helm/cluster-eks/templates/cilium-helmrelease.yaml:19]: will change once the cilium PR is merged https://github.com/giantswarm/cilium-app/pull/99

File: helm/cluster-eks/templates/coredns-helmrelease.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ spec:
       chart: coredns-app
       # used by renovate
       # repo: giantswarm/coredns-app
-      version: 1.19.0
+      version: 1.17.0
```
Comment (whites11) [inline helm/cluster-eks/templates/coredns-helmrelease.yaml:19]: downgrade?

Comment (calvix) [reply helm/cluster-eks/templates/coredns-helmrelease.yaml:19]: https://github.com/giantswarm/giantswarm/issues/28646

**Review by whites11 — APPROVED** [review]
your PRs are like the bible for me.

**Pull request merged**